mlir/include/mlir/Query/Matcher/MatchersInternal.h

```diff
@@ -20,18 +21,50 @@
 #ifndef MLIR_TOOLS_MLIRQUERY_MATCHER_MATCHERSINTERNAL_H
 #define MLIR_TOOLS_MLIRQUERY_MATCHER_MATCHERSINTERNAL_H
 
-#include "mlir/IR/Matchers.h"
+#include "mlir/Analysis/SliceAnalysis.h"
```

Comment:
Could this be in a different file? (have internal only be matcher internal details independent of any matchers)

Reply:
(I don't see any use here TBH)